### PR TITLE
fix: wire up cache

### PR DESCRIPTION
**Pull request by fibonacci1729**
~DEFINITELY WIP~

Implements #1188

### CODE REVIEW

**Comment by fibonacci1729** [comment]
@itowlson putting this up now; would appreciate early eyes. Thanks!

File: crates/trigger/src/cli.rs [hunk 1]
```diff
@@ -96,6 +97,9 @@ where
     /// Load the application from the registry.
     #[clap(long = "from-registry", hide = true)]
     pub from_registry: bool,
+    // /// The root directory of the registry cache.
+    // #[clap(long = "registry-cache-root", hide = true)]
+    // pub registry_cache_root: Option<PathBuf>,
```
Comment (fibonacci1729) [inline crates/trigger/src/cli.rs:102]: I was lazy here...It was easier to pass this from `spin up` using an environment variable....

**Comment by itowlson** [comment]
Acknowledging that this is WIP but since you asked for eyes... the core logic looks sound (though Radu will have to comment on the registry loader code paths), but I was a bit puzzled by the UI.  I had thought the import of the request was to use the existing registry cache directory - which we must already know in order to do `spin up --from-registry` - for URL components as well.  So I'm not sure if we should be adding a new flag to `spin up` to set that cache directory; and if we are, it seems like it's no longer purely a registry cache, more a General Remote Stuff cache?

But I'm not very familiar with this stuff and may be the wrong pair of eyes.  My main role in all this was copying and pasting a TODO comment from an old PR into a new issue...

**Comment by fibonacci1729** [comment]
cc @radu-matei

**Pull request closed without merging**